src/components/icon/icon.test.js

```diff
@@ -0,0 +1,237 @@
+      ["rounded-rect", 20],
+      ["square", 0],
+    ])("should check bgShape as %s for Icon component", (bgShape, radius) => {
+      CypressMountWithProviders(<IconComponent bgShape={bgShape} />);
+      icon()
+        .should("have.css", "border-bottom-left-radius", `${radius}%`)
+        .and("have.css", "border-bottom-right-radius", `${radius}%`)
+        .and("have.css", "border-top-left-radius", `${radius}%`)
+        .and("have.css", "border-top-right-radius", `${radius}%`);
+    });
+
+    it.each([
+      ["small", 24],
+      ["medium", 32],
+      ["large", 40],
+      ["extra-large", 56],
+    ])("should check %s fontSize for Icon component", (fontSize, pixelSize) => {
+      CypressMountWithProviders(<IconComponent fontSize={fontSize} />);
+      icon()
+        .should("have.css", "height", `${pixelSize}px`)
+        .and("have.css", "width", `${pixelSize}px`);
+    });
+
+    it.each(colorData)(
+      "should check %s background color for Icon component",
+      (backgroundColor) => {
+        CypressMountWithProviders(<IconComponent bg={backgroundColor} />);
+        icon().should("have.css", "background-color", backgroundColor);
+      }
+    );
+
+    it.each(colorData)(
+      "should check icon color as %s for Icon component",
+      (iconColor) => {
+        CypressMountWithProviders(<IconComponent color={iconColor} />);
+        icon().should("have.css", "color", iconColor);
+      }
+    );
+
+    it.each([
+      [true, "rgba(0, 0, 0, 0.3)"],
+      [false, "rgba(0, 0, 0, 0.9)"],
+    ])("should check when disabled is %s for Icon component", (bool, color) => {
+      CypressMountWithProviders(<IconComponent disabled={bool} />);
+      icon().should("have.css", "color", color);
+    });
+
+    it.each(testData)(
+      "should check ariaLabel as %s for Icon component",
+      (ariaLabel) => {
+        CypressMountWithProviders(<IconComponent ariaLabel={ariaLabel} />);
+        icon().should("have.attr", "aria-label", ariaLabel);
+      }
+    );
+
+    it.each(["true", "false"])(
+      "should check when aria-hidden is %s for Icon component",
+      (bool) => {
+        CypressMountWithProviders(<IconComponent aria-hidden={bool} />);
+        icon().should("have.attr", "aria-hidden", bool);
+      }
+    );
+
+    it.each(testData)(
+      "should check tooltipMessage as %s for Icon component",
+      (tooltipMessage) => {
+        CypressMountWithProviders(
+          <IconTooltipComponent tooltipMessage={tooltipMessage} />
+        );
+        icon().realHover();
+        getDataElementByValue("tooltip")
+          .should("be.visible")
+          .contains(tooltipMessage);
+      }
+    );
+
+    it.each(["bottom", "left", "right", "top"])(
+      "should check %s position of tooltip for Icon component",
+      (position) => {
+        CypressMountWithProviders(
+          <IconTooltipComponent tooltipPosition={position} />
+        );
+        icon().should("be.visible").and("have.css", position);
+      }
```

Comment:
This verification isn't verifying the tooltip position.  If I change the verification to icon().should("be.visible").and("have.css", "bottom") it will still pass if the tooltip position is set to top.